token/program-2022/src/processor.rs

```diff
@@ -7524,6 +7532,44 @@ mod tests {
             ),
             Err(ProgramError::IncorrectProgramId)
         );
+
+        // Invalid Extension Type for mint and uninitialized account
+        assert_eq!(
+            do_process_instruction(
+                get_account_data_size(
+                    &program_id,
+                    &mint_key,
+                    &[ExtensionType::Uninitialized]
+                )
+                .unwrap(),
+                vec![&mut invalid_mint_account],
+            ),
+            Err(TokenError::ExtensionTypeMismatch.into())
+        );
+        assert_eq!(
+            do_process_instruction(
+                get_account_data_size(
+                    &program_id,
+                    &mint_key,
+                    &[ExtensionType::MintCloseAuthority]
+                )
+                .unwrap(),
+                vec![&mut invalid_mint_account],
+            ),
+            Err(TokenError::ExtensionTypeMismatch.into())
+        );
+        assert_ne!(
+            do_process_instruction(
+                get_account_data_size(
+                    &program_id,
+                    &mint_key,
+                    &[ExtensionType::MemoTransfer]
+                )
+                .unwrap(),
+                vec![&mut invalid_mint_account],
+            ),
+            Err(TokenError::ExtensionTypeMismatch.into())
+        );
```

Comment:
Sorry, I wasn't clear, I meant to mix a valid account extension with an invalid mint extension, ie to do `&[ExtensionType::MemoTransfer, ExtensionType::MintCloseAuthority]`.

Also, these tests are using an invalid mint, so could you move these tests to after 7483 and use the valid mint?

Reply:
yes sure